scheduling/tasks_test.go

```diff
@@ -0,0 +1,148 @@
+package scheduling
+
+import (
+	"testing"
+	"time"
+)
+
+func TestSerializeTaskMetadata(t *testing.T) {
+	testCases := []struct {
+		name string
+		task TaskMetadata
+	}{
+		{
+			name: "WithProviderTarget",
+			task: TaskMetadata{
+				ID:   1,
+				Name: "provider_task",
+				Type: HealthCheck,
+				Target: Provider{
+					Name:       "postgres",
+					TargetType: ProviderTarget,
+				},
+				Date: time.Now(),
+			},
+		},
+		{
+			name: "WithNameVariantTarget",
+			task: TaskMetadata{
+				ID:   1,
+				Name: "nv_task",
+				Type: ResourceCreation,
+				Target: NameVariant{
+					Name:       "transaction",
+					TargetType: NameVariantTarget,
+				},
+				Date: time.Now(),
+			},
+		},
+	}
+
+	for _, currTest := range testCases {
+		t.Run(currTest.name, func(t *testing.T) {
+			serializeTask, err := currTest.task.ToJSON()
+			if err != nil {
+				t.Errorf("failed to serialize task metadata: %v", err)
```

Comment:
Should be t.Fatal. Fatal should be used everytime something after this depends on a value, otherwise you start passing around nil pointers since Error lets the test continue